ur_robot_driver/src/ros/hardware_interface.cpp

```diff
@@ -315,6 +315,19 @@ bool HardwareInterface::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw
   // Calling this service will make the "External Control" program node on the UR-Program return.
   deactivate_srv_ = robot_hw_nh.advertiseService("hand_back_control", &HardwareInterface::stopControl, this);
 
+  // Calling this service will zero the robot's ftsensor. Note: On e-Series robots this will only
+  // work when the robot is in remote-control mode.
+  tare_sensor_srv_ = robot_hw_nh.advertiseService<std_srvs::Trigger::Request, std_srvs::Trigger::Response>(
+      "zero_ftsensor", [&](std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp) {
+        resp.success = this->ur_driver_->sendScript(R"(sec tareSensor():
+  set_digital_out(1,True)
```

Comment:
I'm not entirely up-to-speed on e-Series controllers: why the `set_digital_out(..)` here? Is the ft sensor connected to it?

Reply:
oops, that was from debugging...

Reply:
fixed.